x-pack/plugins/ml/public/application/components/ml_inference/components/on_failure_configuration.tsx

```diff
@@ -0,0 +1,220 @@
+                      <EuiFlexItem grow={false}>
+                        <EuiButtonEmpty
+                          iconType="pencil"
+                          size="xs"
+                          onClick={() => {
+                            setEditOnFailure(!editOnFailure);
+                          }}
+                        >
+                          {editOnFailure ? cancelEditMessage : editMessage}
+                        </EuiButtonEmpty>
+                      </EuiFlexItem>
+                      <EuiFlexItem grow={false}>
+                        {editOnFailure ? (
+                          <SaveChangesButton
+                            onClick={updateOnFailure}
+                            disabled={isOnFailureValid === false}
+                          />
+                        ) : null}
+                      </EuiFlexItem>
+                      <EuiFlexItem grow={false}>
+                        {editOnFailure ? (
+                          <EuiButtonEmpty size="xs" onClick={resetOnFailure}>
+                            {i18n.translate(
+                              'xpack.ml.trainedModels.content.indices.pipelines.addInferencePipelineModal.steps.advanced.resetOnFailureButton',
+                              { defaultMessage: 'Reset' }
+                            )}
+                          </EuiButtonEmpty>
+                        ) : null}
+                      </EuiFlexItem>
+                    </EuiFlexGroup>
+                  }
+                  helpText={
+                    <FormattedMessage
+                      id="xpack.ml.trainedModels.content.indices.pipelines.addInferencePipelineModal.steps.advanced.onFailureHelpText"
+                      defaultMessage="In the case of failure, the above configuration will set the index document as failed, provide the timestamp at which ingest failed, and provide context for the failure."
+                    />
+                  }
+                >
+                  <>
+                    {!editOnFailure ? (
+                      <EuiCodeBlock isCopyable={true} overflowHeight={350}>
+                        {JSON.stringify(onFailure, null, 2)}
+                      </EuiCodeBlock>
+                    ) : null}
+                    {editOnFailure ? (
+                      <>
+                        <EuiSpacer size="s" />
+                        <CodeEditor
+                          height={300}
+                          languageId="json"
+                          options={{
+                            automaticLayout: true,
+                            lineNumbers: 'off',
+                            tabSize: 2,
+                          }}
+                          value={onFailureString}
+                          onChange={handleOnFailureChange}
+                        />
+                      </>
+                    ) : null}
+                  </>
+                </EuiFormRow>
+              </EuiFlexItem>
+              <EuiFlexItem>
```

Comment:
I wonder if the order of these two sections should be swapped, so that the switch appears first? The help text to the left refers to the `ignore_failure` first.

Reply:
I agree - swapped in b0aa5e0d8ecc48b474dcd709982536fe852dcd9a